{"ast":null,"code":"import React,{Component}from'react';import{jsx as _jsx}from\"react/jsx-runtime\";class ErrorBoundary extends Component{constructor(props){super(props);this.state={hasError:false};}componentDidCatch(error,errorInfo){console.error('Error caught by error boundary:',error,errorInfo);this.setState({hasError:true});}render(){if(this.state.hasError){return/*#__PURE__*/_jsx(\"h1\",{children:\"Something went wrong.\"});}return this.props.children;}}export default ErrorBoundary;","map":{"version":3,"names":["React","Component","jsx","_jsx","ErrorBoundary","constructor","props","state","hasError","componentDidCatch","error","errorInfo","console","setState","render","children"],"sources":["/home/falrabadi/workspace/projects/jns/frontend/src/ErrorBoundary.js"],"sourcesContent":["import React, { Component } from 'react';\n\nclass ErrorBoundary extends Component {\n  constructor(props) {\n    super(props);\n    this.state = { hasError: false };\n  }\n\n  componentDidCatch(error, errorInfo) {\n    console.error('Error caught by error boundary:', error, errorInfo);\n    this.setState({ hasError: true });\n  }\n\n  render() {\n    if (this.state.hasError) {\n      return <h1>Something went wrong.</h1>;\n    }\n\n    return this.props.children; \n  }\n}\n\nexport default ErrorBoundary;\n"],"mappings":"AAAA,MAAO,CAAAA,KAAK,EAAIC,SAAS,KAAQ,OAAO,CAAC,OAAAC,GAAA,IAAAC,IAAA,yBAEzC,KAAM,CAAAC,aAAa,QAAS,CAAAH,SAAU,CACpCI,WAAWA,CAACC,KAAK,CAAE,CACjB,KAAK,CAACA,KAAK,CAAC,CACZ,IAAI,CAACC,KAAK,CAAG,CAAEC,QAAQ,CAAE,KAAM,CAAC,CAClC,CAEAC,iBAAiBA,CAACC,KAAK,CAAEC,SAAS,CAAE,CAClCC,OAAO,CAACF,KAAK,CAAC,iCAAiC,CAAEA,KAAK,CAAEC,SAAS,CAAC,CAClE,IAAI,CAACE,QAAQ,CAAC,CAAEL,QAAQ,CAAE,IAAK,CAAC,CAAC,CACnC,CAEAM,MAAMA,CAAA,CAAG,CACP,GAAI,IAAI,CAACP,KAAK,CAACC,QAAQ,CAAE,CACvB,mBAAOL,IAAA,OAAAY,QAAA,CAAI,uBAAqB,CAAI,CAAC,CACvC,CAEA,MAAO,KAAI,CAACT,KAAK,CAACS,QAAQ,CAC5B,CACF,CAEA,cAAe,CAAAX,aAAa","ignoreList":[]},"metadata":{},"sourceType":"module","externalDependencies":[]}